exporters/cinder.go

```diff
@@ -105,16 +109,31 @@ func ListVolumes(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metric) e
 	ch <- prometheus.MustNewConstMetric(exporter.Metrics["volumes"].Metric,
 		prometheus.GaugeValue, float64(len(allVolumes)))
 
-	// Volume status metrics
+	// Volume_gb metrics
 	for _, volume := range allVolumes {
 		if volume.Attachments != nil && len(volume.Attachments) > 0 {
-			ch <- prometheus.MustNewConstMetric(exporter.Metrics["volume_status"].Metric,
-				prometheus.GaugeValue, float64(mapVolumeStatus(volume.Status)), volume.ID, volume.Name,
-				volume.Status, volume.Bootable, volume.TenantID, strconv.Itoa(volume.Size), volume.VolumeType, volume.Attachments[0].ServerID)
+			ch <- prometheus.MustNewConstMetric(exporter.Metrics["volume_gb"].Metric,
+				prometheus.GaugeValue, float64(volume.Size), volume.ID, volume.Name,
+				volume.Status, volume.Bootable, volume.TenantID, volume.VolumeType, volume.Attachments[0].ServerID)
 		} else {
-			ch <- prometheus.MustNewConstMetric(exporter.Metrics["volume_status"].Metric,
-				prometheus.GaugeValue, float64(mapVolumeStatus(volume.Status)), volume.ID, volume.Name,
-				volume.Status, volume.Bootable, volume.TenantID, strconv.Itoa(volume.Size), volume.VolumeType, "")
+			ch <- prometheus.MustNewConstMetric(exporter.Metrics["volume_gb"].Metric,
+				prometheus.GaugeValue, float64(volume.Size), volume.ID, volume.Name,
+				volume.Status, volume.Bootable, volume.TenantID, volume.VolumeType, "")
+		}
+	}
+
+	// Volume status metrics
+	if !exporter.ExporterConfig.DisableDeprecatedMetrics {
```

Comment:
I think the correct way for this would be to move this metric into a separated function and simply let the if exporter.isDeprecatedMetric to check for this condition.